pkg/minikube/bootstrapper/kubeadm/kubeadm.go

```diff
@@ -658,6 +664,17 @@ func (k *Bootstrapper) restartControlPlane(cfg config.ClusterConfig) error {
 		}
 	}
 
+	if cfg.VerifyComponents[kverify.OperationalKey] {
+		// after kubelet is restarted (with 'kubeadm init phase kubelet-start' above),
+		// it appears to be immediately Ready as are all kube-system pods
+		// then (after ~10sec) it realises it has some changes to apply, implying also pods restarts
+		// so we wait for kubelet to initialise itself...
+		time.Sleep(10 * time.Second)
```

Comment:
I dont think this is the right thing to sleep 10 seconds randomly, is there any other way to detect that kubelet is trying to initialize itself ?

Reply:
you are right @medyagh and thank you for proposing to use retry.Expo() instead